src/argilla/client/feedback/schemas.py

```diff
@@ -93,51 +93,165 @@ def user_id_must_have_value(cls, v):
+    question_name: str
+    type: Optional[Literal["model", "human"]] = None
+    score: Optional[float] = None
+    value: Any
+    agent: Optional[str] = None
+
+    class Config:
+        extra = Extra.forbid
+
+
+class FeedbackRecord(BaseModel):
+    """Schema for the records of a `FeedbackDataset` in Argilla.
+
+    Args:
+        id: The ID of the record in Argilla. Defaults to None, and is automatically
+            fulfilled internally once the record is pushed to Argilla.
+        fields: Fields that match the `FeedbackDataset` defined fields. So this attribute
+            contains the actual information shown in the UI for each record, being the
+            record itself.
+        metadata: Metadata to be included to enrich the information for a given record.
+            Note that the metadata is not shown in the UI so you'll just be able to see
+            that programatically after pulling the records. Defaults to None.
+        responses: Responses given by either the current user, or one or a collection of
+            users that must exist in Argilla. Each response corresponds to one of the
+            `FeedbackDataset` questions, so the values should match the question type.
+            Defaults to None.
+        external_id: The external ID of the record, which means that the user can
+            specify this ID to identify the record no matter what the Argilla ID is.
+            Defaults to None.
+
+    Examples:
+        >>> from argilla.client.feedback.schemas import FeedbackRecord, ResponseSchema, ValueSchema
+        >>> FeedbackRecord(
         ...     fields={"text": "This is the first record", "label": "positive"},
         ...     metadata={"first": True, "nested": {"more": "stuff"}},
-        ...     responses=[rg.ResponseSchema(values={"question-1": {"value": "This is the first answer"}, "question-2": {"value": 5}}))],
+        ...     responses=[ # optional
+        ...         ResponseSchema(
+        ...             user_id="user-1",
+        ...             values={
+        ...                 "question-1": ValueSchema(value="This is the first answer"),
+        ...                 "question-2": ValueSchema(value=5),
+        ...             },
+        ...             status="submitted",
+        ...         ),
+        ...     ],
+        ...     suggestions=[ # optional
+        ...         SuggestionSchema(
+        ...            question_name="question-1",
+        ...            type="model",
+        ...            score=0.9,
+        ...            value="This is the first suggestion",
+        ...            agent="agent-1",
+        ...         ),
         ...     external_id="entry-1",
         ... )
 
     """
 
+    id: Optional[UUID] = None
     fields: Dict[str, str]
-    metadata: Optional[Dict[str, Any]] = None
-    responses: Optional[Union[ResponseSchema, List[ResponseSchema]]] = None
+    metadata: Optional[Dict[str, Any]] = Field(default_factory=dict)
```

Comment:
Yes it's just for that and so that anyone can just `record.metadata.update({...: ...})` but we can make it None if you want (also we're excluding the unset values with `exclude_unset` when calling `dict()` so if that's an empty dict we'll just push `None`)